ext/intl/bidi/bidi.c

```diff
@@ -330,32 +353,84 @@ static PHP_METHOD(IntlBidi, setPara) {
 }
 /* }}} */
 
-/* {{{ proto self IntlBidi::setLine(int $tart, int $limit, IntlBidi $line) */
-ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(bidi_setline_arginfo, ZEND_RETURN_VALUE, 3, IS_OBJECT, 0)
+/* {{{ proto IntlBidi IntlBidi::setLine(int $start, int $limit) */
+ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(bidi_setline_arginfo, ZEND_RETURN_VALUE, 2, IS_OBJECT, 0)
 	ZEND_ARG_TYPE_INFO(0, start, IS_LONG, 0)
 	ZEND_ARG_TYPE_INFO(0, limit, IS_LONG, 0)
-	ZEND_ARG_TYPE_INFO(0, line, IS_OBJECT, 0)
 ZEND_END_ARG_INFO();
 static PHP_METHOD(IntlBidi, setLine) {
 	zend_long start, limit;
-	zend_object *zline;
-	php_intl_bidi_object *objval = bidi_object_from_zend_object(Z_OBJ_P(getThis()));
-	php_intl_bidi_object *lineval;
+	zval retval;
+	zend_function * constructor;
+	zend_fcall_info fci;
+	zend_fcall_info_cache fcc;
+	php_intl_bidi_object *objval, *lineval;
 	UErrorCode error;
 
-	if (zend_parse_parameters_throw(ZEND_NUM_ARGS(), "llO", &start, &limit, &zline, php_intl_bidi_ce) == FAILURE) {
-		return;
+	// Parse parameters.
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_LONG(start)
+		Z_PARAM_LONG(limit)
+	ZEND_PARSE_PARAMETERS_END();
+
+	// init return value as new object. (This code is from php_reflection ReflectionClass::newInstance())
+
+	// init the return value as a IntlBidi instance.
+	object_init_ex(return_value, php_intl_bidi_ce);
+
+
+	// get the constructor.
+	constructor = Z_OBJ_HT_P(return_value)->get_constructor(Z_OBJ_P(return_value));
```

Comment:
So this is technically correct (the best kind of correct), but I will say that you don't need to call the constructor via the Zend callstack.  Instead, you could pull out the interesting bits from the __construct method above into a C helper function, then invoke that with the objval and args you want.  It'll be a lot more readable and a little more performant.

Reply:
Changed that